src/utils/addr.rs

```diff
@@ -0,0 +1,71 @@
+use std::borrow::Cow;
+use std::fmt;
+use std::str::FromStr;
+
+use serde::de::{self, Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+use thiserror::Error;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum AddrMode {
+    Abs,
+    ModRel(usize),
```

Comment:
_nit_: Could be `Rel` or `Relative` since this is also `"rel"` in the string representation.